src/HDF5.jl

```diff
@@ -1378,26 +1376,41 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
+# Helper for reading compound types
+function read_row(io::IO, membertype, membersize)
+    row = Vector{Any}()
```

Comment:
I think `@tkelman` mentioned that there have been some discussions of deprecated this type of constructor. May be best to use `Any[]` to be safe